### PR TITLE
feat(cli): add --trait flag to item add command

### PR DESCRIPTION
**Pull request by chapel**
## Summary
- Add `--trait <trait...>` option to `kspec item add` to apply traits at creation time
- Supports multiple traits via repeated `--trait` flags
- Updated help text with trait usage example
- Saves a command when creating specs with known traits

## Test plan
- [x] Added 4 tests covering single/multiple traits and persistence
- [x] All 25 trait-cli tests pass
- [x] Manual verification with `kspec item add --help`

Task: @01KGTXZ3

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/cli/commands/item.ts [hunk 1]
```diff
@@ -650,7 +652,7 @@ Examples:
           implements: [],
           relates_to: [],
           tests: [],
-          traits: [],
+          traits: options.trait || [],
```
Comment (chapel) [inline src/cli/commands/item.ts]: MUST-FIX: `traits: options.trait || []` bypasses trait resolution/validation and accepts invalid refs as success. Repro (using current branch): `item add --under @test-core --title "Bad trait" --trait @does-not-exist --json` exits 0 and persists `traits:["@does-not-exist"]`; `--trait @test-core` (non-trait item) also exits 0. This should match `item trait add` semantics: resolve each ref, require existence, enforce `type === "trait"`, and store canonical refs.

**Review by chapel — COMMENTED** [review]
Automated review findings (REQUEST_CHANGES intent, posted as COMMENT because GitHub disallows requesting changes on your own PR).

Found one MUST-FIX correctness issue and one SHOULD-FIX test gap.

Task linkage check: PR body includes `Task: @01KGTXZ3`, but that task currently has `spec_ref: null`, so there are no linked spec ACs to validate for annotation coverage in this review.

File: tests/trait-cli.test.ts [hunk 2]
```diff
@@ -488,3 +488,64 @@ features:
...
+  it('should create spec item with trait applied', async () => {
+    const result = kspecJson<{ item: { traits: string[] } }>(
+      'item add --under @test-core --title "Feature with trait" --trait @json-output',
+      tempDir
+    );
+
+    expect(result.item.traits).toContain('@json-output');
+  });
+
+  it('should apply multiple traits', async () => {
```
Comment (chapel) [inline tests/trait-cli.test.ts]: SHOULD-FIX: the new suite only covers happy paths and no-trait default. Please add negative tests for invalid trait refs and non-trait refs (expected non-zero exit with clear errors), plus an edge case for canonicalization/dedup when the same trait is passed via different ref forms.
